etcdserver/server.go

```diff
@@ -280,14 +286,14 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 			raftStorage: s,
 			storage:     NewStorage(w, ss),
 		},
-		id:             id,
-		attributes:     Attributes{Name: cfg.Name, ClientURLs: cfg.ClientURLs.StringSlice()},
-		Cluster:        cfg.Cluster,
-		stats:          sstats,
-		lstats:         lstats,
-		SyncTicker:     time.Tick(500 * time.Millisecond),
-		reqIDGen:       idutil.NewGenerator(uint8(id), time.Now()),
-		clusterVersion: semver.Must(semver.NewVersion(version.MinClusterVersion)),
+		id:            id,
+		attributes:    Attributes{Name: cfg.Name, ClientURLs: cfg.ClientURLs.StringSlice()},
+		Cluster:       cfg.Cluster,
+		stats:         sstats,
+		lstats:        lstats,
+		SyncTicker:    time.Tick(500 * time.Millisecond),
+		reqIDGen:      idutil.NewGenerator(uint8(id), time.Now()),
+		forceVersionC: make(chan struct{}),
```

Comment:
i think `make(chan struct{}, 1)` is better, so server main loop is blocked by forceVersion.